src/sqliteformat.cpp

```diff
@@ -560,72 +560,84 @@ bool SqliteFormat::modifyComponents(const Incidence::Ptr &incidence, const QStri
+        }
         SL3_bind_int64(stmt1, index, secs);
 
-        SL3_bind_int(stmt1, index, incidence->revision());
+        SL3_bind_int(stmt1, index, incidence.revision());
 
-        comments = incidence->comments().join(" ").toUtf8();
+        comments = incidence.comments().join(" ").toUtf8();
         SL3_bind_text(stmt1, index, comments.constData(), comments.length(), SQLITE_STATIC);
 
         // Attachments are now stored in a dedicated table.
         SL3_bind_text(stmt1, index, nullptr, 0, SQLITE_STATIC);
 
-        contact = incidence->contacts().join(" ").toUtf8();
+        contact = incidence.contacts().join(" ").toUtf8();
         SL3_bind_text(stmt1, index, contact.constData(), contact.length(), SQLITE_STATIC);
 
         // Never save recurrenceId as FLOATING_DATE, because the time of a
         // floating date is not guaranteed on read and recurrenceId is used
         // for date-time comparisons.
-        SL3_bind_date_time(this, stmt1, index, incidence->recurrenceId(), false);
+        SL3_bind_date_time(this, stmt1, index, incidence.recurrenceId(), false);
 
-        relatedtouid = incidence->relatedTo().toUtf8();
+        relatedtouid = incidence.relatedTo().toUtf8();
         SL3_bind_text(stmt1, index, relatedtouid.constData(), relatedtouid.length(), SQLITE_STATIC);
 
-        url = incidence->url().toString().toUtf8();
+        url = incidence.url().toString().toUtf8();
         SL3_bind_text(stmt1, index, url.constData(), url.length(), SQLITE_STATIC);
 
-        uid = incidence->uid().toUtf8();
+        uid = incidence.uid().toUtf8();
         SL3_bind_text(stmt1, index, uid.constData(), uid.length(), SQLITE_STATIC);
 
-        if (incidence->type() == Incidence::TypeEvent) {
-            Event::Ptr event = incidence.staticCast<Event>();
+        if (incidence.type() == Incidence::TypeEvent) {
+            const Event *event = static_cast<const Event*>(&incidence);
             SL3_bind_int(stmt1, index, (int)event->transparency());
         } else {
             SL3_bind_int(stmt1, index, 0);
         }
 
-        SL3_bind_int(stmt1, index, (int) incidence->localOnly());
+        SL3_bind_int(stmt1, index, (int) incidence.localOnly());
 
         int percentComplete = 0;
         QDateTime effectiveDtCompleted;
-        if (incidence->type() == Incidence::TypeTodo) {
-            Todo::Ptr todo = incidence.staticCast<Todo>();
+        if (incidence.type() == Incidence::TypeTodo) {
+            const Todo *todo = static_cast<const Todo*>(&incidence);
             percentComplete = todo->percentComplete();
             if (todo->isCompleted()) {
                 if (!todo->hasCompletedDate()) {
                     // If the todo was created by KOrganizer<2.2 it does not have
-                    // a correct completion date. Set one now.
-                    todo->setCompleted(QDateTime::currentDateTimeUtc());
+                    // a correct completion date.
```

Comment:
KOrganizer 2.2, eh? Seems released April 2001, hardly relevant anymore. Maybe the code wouldn't be too bad to protect against missing a value, but at least the reference could be dropped. Or optionally just skip all this special logic altogher, nothing is using todos now anyway.